src/databricks/labs/blueprint/paths.py

```diff
@@ -296,6 +144,16 @@ def __init__(self, ws: WorkspaceClient, *args) -> None:  # pylint: disable=super
         self._path_parts = path_parts
         self._ws = ws
 
+    @classmethod
+    def _from_object_info(cls, ws: WorkspaceClient, object_info: ObjectInfo):
+        """Special (internal-only) constructor that creates an instance based on ObjectInfo."""
+        if not object_info.path:
+            msg = f"Cannot initialise within object path: {object_info}"
+            raise ValueError(msg)
+        path = cls(ws, object_info.path)
+        path._cached_object_info = object_info
```

Comment:
this is a nice performance optimization compared to getting status for every path